src/test/sigopcount_tests.cpp

```diff
@@ -64,4 +65,16 @@ BOOST_AUTO_TEST_CASE(GetSigOpCount)
     BOOST_CHECK_EQUAL(p2sh.GetSigOpCount(scriptSig2), 3U);
 }
 
+
+BOOST_AUTO_TEST_CASE(blockSigOpAcceptLimit)
+{
+    BOOST_CHECK_EQUAL(Policy::blockSigOpAcceptLimit(0), 20000);
+    BOOST_CHECK_EQUAL(Policy::blockSigOpAcceptLimit(100), 20000);
+    BOOST_CHECK_EQUAL(Policy::blockSigOpAcceptLimit(50000), 20000);
+    BOOST_CHECK_EQUAL(Policy::blockSigOpAcceptLimit(1e6), 20000);
+    BOOST_CHECK_EQUAL(Policy::blockSigOpAcceptLimit(1e6 + 1), 40000);
+    BOOST_CHECK_EQUAL(Policy::blockSigOpAcceptLimit(2e6), 40000);
+    BOOST_CHECK_EQUAL(Policy::blockSigOpAcceptLimit(4e6 + 1), 100000);
```

Comment:
It in unclear why these test case are chosen rather than others.

A few things. First 1e6 is a floating point, and you don't want to use that. This is annoying, but 10000000 is the way to go. Create a constant if that is too constraining. It's probably fine in that range, but test with larger value and you notice that 1eN + 1 == 1eN for larges values of N.

The exact value of N will depend of the rounding mode the CPU is in when running the code and the precision that the compiler chose to use, both of which are non standard.

I would test for a set of  X*1000000, X*1000000+1 and X*1000000-1 for various values of X, typical edges cases values (0, 1, uint.max)  and then some randomly picked values.